core/filter.js

```diff
@@ -165,11 +175,12 @@
 		// Register filter instance.
 		CKEDITOR.filter.instances[ this.id ] = this;
 
-		if ( editorOrRules instanceof CKEDITOR.editor ) {
-			var editor = this.editor = editorOrRules;
+		this.editor = editorOrRules instanceof CKEDITOR.editor ? editorOrRules : null;
```

Comment:
Why not:

```javascript
var editor = this.editor = editorOrRules instanceof CKEDITOR.editor ? editorOrRules : null;
```
? With such version there won't be need to change all references to the editor.